test/e2e/run.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash -e
+
+export NOTATION_E2E_BINARY_PATH=$(realpath $1)
+if [ ! -f "$NOTATION_E2E_BINARY_PATH" ];then
+    echo "run.sh <notation-binary-path>"
+    exit 1
+fi
+
+# install dependency
+go install -mod=mod github.com/onsi/ginkgo/v2/ginkgo
+
+# set environment variable for E2E testing
+REG_HOST=localhost
+REG_PORT=5000
+ZOT_CONTAINER_NAME=zot
```

Comment:
- Yes, Zot support OCI artifact and OCI layout. Currently, we cannot use another registry directly, because we use OCI layout to store the repository data for testing and it depends on the file system. If we want to test other registries, we need oras OCI layout feature to copy the OCI layout files to the registry.
- run.sh is for local testing, we can rename it to be local.sh. For ACR testing, we need a new script to create the registry before test and delete it after test, so the pipeline can run multi-registry test in parallel. However, multi-registry support will not be covered in this PR.

Reply:
Update:

Refactored the run.sh and the registry logic has been moved to ./scripts/zot.sh

Reply:
Update:
Added DockerHub support!